compactor/src/lib.rs

```diff
@@ -109,12 +113,12 @@ async fn compact_candidates_with_memory_budget<C, Fut>(
                 warn!(
                     %e,
                     ?partition_id,
-                    compaction_type,
+                    compaction_type_string,
```

Comment:
```suggestion
                    %compaction_type,
```

And implement `std::fmt::Display` for `CompactionType`. No need to manually convert it to a string.

Reply:
Done